--- origsrc/eog-plugins-2.30.1/configure.ac	2010-06-25 05:38:59.797694900 -0500
+++ src/eog-plugins-2.30.1/configure.ac	2010-06-25 05:14:13.542686000 -0500
@@ -361,7 +361,7 @@ AC_SUBST([BUILD_PLUGINS],[$PLUGINS])
 # ****
 AC_PATH_PROG(GLIB_GENMARSHAL, glib-genmarshal)
 
-PLUGIN_LIBTOOL_FLAGS="-module -avoid-version"
+PLUGIN_LIBTOOL_FLAGS="-module -avoid-version -no-undefined"
 AC_SUBST(PLUGIN_LIBTOOL_FLAGS)
 
 AC_CONFIG_FILES([
--- origsrc/eog-plugins-2.30.1/plugins/exif-display/Makefile.am	2010-06-25 05:38:59.804695300 -0500
+++ src/eog-plugins-2.30.1/plugins/exif-display/Makefile.am	2010-06-25 05:14:13.606689700 -0500
@@ -16,7 +16,7 @@ libexif_display_la_SOURCES = \
 	eog-exif-display-plugin.c
 
 libexif_display_la_LDFLAGS = \
-	-avoid-version -module
+	$(PLUGIN_LIBTOOL_FLAGS)
 
 libexif_display_la_LIBADD = $(EOG_LIBS)			\
 	$(EXIFDISPLAY_LIBS)
--- origsrc/eog-plugins-2.30.1/plugins/fit-to-width/Makefile.am	2010-06-25 05:38:59.808695600 -0500
+++ src/eog-plugins-2.30.1/plugins/fit-to-width/Makefile.am	2010-06-25 05:14:13.646692000 -0500
@@ -14,7 +14,7 @@ libfit_to_width_la_SOURCES = \
 	eog-fit-to-width-plugin.c
 
 libfit_to_width_la_LDFLAGS = \
-	-avoid-version -module
+	$(PLUGIN_LIBTOOL_FLAGS)
 
 libfit_to_width_la_LIBADD = $(EOG_LIBS)
 
--- origsrc/eog-plugins-2.30.1/plugins/send-by-mail/Makefile.am	2010-01-09 15:44:12.000000000 -0600
+++ src/eog-plugins-2.30.1/plugins/send-by-mail/Makefile.am	2010-06-25 05:36:31.482211800 -0500
@@ -14,7 +14,7 @@ libsend_by_mail_la_SOURCES = \
 	eog-send-by-mail-plugin.c
 
 libsend_by_mail_la_LDFLAGS = \
-	-avoid-version -module
+	$(PLUGIN_LIBTOOL_FLAGS)
 
 libsend_by_mail_la_LIBADD = $(EOG_LIBS)
 
